packages/grafana-ui/src/components/DateTimePickers/DateTimePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,222 @@
+      setOpen(true);
+    },
+    [setOpen]
+  );
+
+  return (
+    <div data-testid="date-time-picker" style={{ position: 'relative' }}>
+      <DateTimeInput date={date} onChange={onChange} isFullscreen={isFullscreen} onOpen={onOpen} label={label} />
+      {isOpen ? (
+        isFullscreen ? (
+          <ClickOutsideWrapper onClick={() => setOpen(false)}>
+            <DateTimeCalendar date={date} onChange={onApply} isFullscreen={true} onClose={() => setOpen(false)} />
+          </ClickOutsideWrapper>
+        ) : (
+          <Portal>
+            <ClickOutsideWrapper onClick={() => setOpen(false)}>
+              <div className={styles.modal} onClick={stopPropagation}>
+                <DateTimeCalendar date={date} onChange={onApply} isFullscreen={false} onClose={() => setOpen(false)} />
+              </div>
+              <div className={containerStyles.backdrop} onClick={stopPropagation} />
+            </ClickOutsideWrapper>
+          </Portal>
+        )
+      ) : null}
+    </div>
+  );
+};
+
+interface DateTimeCalendarProps {
+  date?: DateTime;
+  onChange: (date: DateTime) => void;
+  onClose: () => void;
+  isFullscreen: boolean;
+}
+
+interface InputProps {
+  label?: ReactNode;
+  date?: DateTime;
+  isFullscreen: boolean;
+  onChange: (date: DateTime) => void;
+  onOpen: (event: FormEvent<HTMLElement>) => void;
+}
+
+type InputState = {
+  value: string;
+  invalid: boolean;
+};
+
+const DateTimeInput: FC<InputProps> = ({ date, label, onChange, isFullscreen, onOpen }) => {
+  const [internalDate, setInternalDate] = useState<InputState>(() => {
+    return { value: date ? dateTimeFormat(date) : dateTimeFormat(dateTime()), invalid: false };
+  });
+
+  useEffect(() => {
+    if (date) {
+      setInternalDate({
+        invalid: !isValid(dateTimeFormat(date)),
+        value: isDateTime(date) ? dateTimeFormat(date) : date,
+      });
+    }
+  }, [date]);
+
+  const onChangeDate = useCallback((event: FormEvent<HTMLInputElement>) => {
+    const isInvalid = !isValid(event.currentTarget.value);
+    setInternalDate({
+      value: event.currentTarget.value,
+      invalid: isInvalid,
+    });
+  }, []);
+
+  const onFocus = useCallback(
+    (event: FormEvent<HTMLElement>) => {
+      if (!isFullscreen) {
+        return;
+      }
+      onOpen(event);
+    },
+    [isFullscreen, onOpen]
+  );
+
+  const onBlur = useCallback(() => {
+    if (isDateTime(internalDate.value)) {
+      onChange(dateTime(internalDate.value));
+    }
+  }, [internalDate.value, onChange]);
+
+  const icon = <Button icon="calendar-alt" variant="secondary" onClick={onOpen} />;
+  return (
+    <Field label={label} onClick={stopPropagation} invalid={internalDate.invalid} error="Incorrect date format">
+      <Input
+        onClick={stopPropagation}
+        onChange={onChangeDate}
+        addonAfter={icon}
+        value={internalDate.value}
+        onFocus={onFocus}
+        onBlur={onBlur}
+        data-testid="date-time-input"
```

Comment:
```suggestion
        data-testid="date-time-input"
        placeholder={'Select date/time'}
```
I think it's good to have a placeholder, which will be shown when there's no date selected.